src/tariff_region.h

```diff
@@ -0,0 +1,111 @@
+#ifndef CYCAMORE_SRC_TARIFF_REGION_H_
+#define CYCAMORE_SRC_TARIFF_REGION_H_
+
+#include "cyclus.h"
+#include <string>
+#include <limits>
+#include <utility>
+#include <map>
+#include <vector>
+
+namespace cycamore {
+
+class TariffRegion : public cyclus::Region {
+ public:
+  TariffRegion(cyclus::Context* ctx);
+  virtual ~TariffRegion();
+
+  virtual void EnterNotify();
+  virtual void Tock();
+
+  // Required DRE Functions
+  virtual void AdjustMatlPrefs(cyclus::PrefMap<cyclus::Material>::type& prefs);
+  virtual void AdjustProductPrefs(cyclus::PrefMap<cyclus::Product>::type& prefs);
+
+ private:
+  // Find the best matching region for a supplier
+  cyclus::Region* FindMatchingRegion(cyclus::Facility* supplier);
+  
+  // Find the appropriate tariff for a given region and commodity
+  // Uses tiered override system: region-specific commodity > region blanket > 
+  // global commodity > global blanket
+  double FindTariffForCommodity(cyclus::Region* region, const std::string& commodity);
+  
+  // Helper: Check if any tariff configuration exists
+  bool HasTariffConfiguration() const;
+  
+  // Helper: Check if configuration has been recorded
+  bool ConfigurationRecorded() const;
+  
+  // Helper: Get region prototype name (reduces repeated code)
+  std::string GetRegionName(cyclus::Region* region) const;
+  
+  // Validate the tariff configuration
+  void ValidateConfiguration();
+  
+  // Record tariff configuration to database
+  void RecordTariffConfiguration();
+
+  #pragma cyclus
+
+  // Template function to reduce code duplication between AdjustMatlPrefs and 
+  // AdjustProductPrefs
+  template<typename T>
+  void AdjustPrefsImpl(typename cyclus::PrefMap<T>::type& prefs);
+  
+  // clang-format off
+  #pragma cyclus var { \
+    "default": {}, \
+    "alias": ["adjustment_regions", "region", ["RegionConfig", "blanket_adjustment", ["commodity_adjustments", "commodity", "adjustment"]]], \
```

Comment:
Recommend calling this "TariffConfig" instead 
```suggestion
    "alias": ["adjustment_regions", "region", ["TariffConfig", "blanket_adjustment", ["commodity_adjustments", "commodity", "adjustment"]]], \
```

Reply:
I can do that! I'll make that change with a few other ones on my end and then push (for some reason the "commit change" on GitHub option scares me)

Reply:
Should be done now!